### PR TITLE
Issue #13345: enable example test for SeverityMatchFilterExamplesTest

### PR DESCRIPTION
**Pull request by Kexin2000**
part of issue: #13345
this is the first example test for filter, I've encountered some issues and need to discuss potential solutions. @romani @nrmancuso 
1. My current approach to writing filter examples is inspired by [this file](https://github.com/checkstyle/checkstyle/blob/571e547a9f7b0a05503b8d76f34181059a8c3bbf/src/xdocs-examples/resources-noncompilable/com/puppycrawl/tools/checkstyle/filters/suppresswithplaintextcommentfilter/Example9.java#L4): demonstrating the function of the filter through other checks. I used ParameterNameCheck and MethodNameCheck, setting the severity of ParameterNameCheck to info, while the MethodNameCheck defaults to error. However, when I run SeverityMatchFilterExamplesTest, the code enters line 382 of the following code block and throws an exception. The reason is that the path where SeverityMatchFilterExamplesTest is located does not contain module name parameters for 'parametername' and 'methodname'. So, I added 'parametername' and 'methodname' to the hashmap 'moduleMappings'. I think this approach should be acceptable, as I used 'git blame' to investigate the historical reason for 'moduleMappings'. The existence of this variable is to address some special cases in #13345, such as SuppressWarningsHolder #13845.

https://github.com/checkstyle/checkstyle/blob/571e547a9f7b0a05503b8d76f34181059a8c3bbf/src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java#L372-L390

2. While running SeverityMatchFilterExamplesTest, there's a similar issue with a path mismatch. The code will throw an exception at line 602 of the following code block because there is no message in the filters package. So, I added a messages.properties file under src/main/resources/com/puppycrawl/tools/checkstyle/filters, and for subsequent filter example tests, more violation messages can be added to this file to resolve this issue. Is my solution feasible? (Messages.properties files for other languages also need to be added.)

https://github.com/checkstyle/checkstyle/blob/571e547a9f7b0a05503b8d76f34181059a8c3bbf/src/test/java/com/puppycrawl/tools/checkstyle/AbstractModuleTestSupport.java#L600-L610

### CODE REVIEW

**Comment by Kexin2000** [comment]
The work on this PR is not yet complete. To avoid unnecessary work, I want to first confirm a suitable solution.

File: src/xdocs/filters/severitymatchfilter.xml [hunk 1]
```diff
@@ -48,13 +48,19 @@
...
       <subsection name="Examples" id="Examples">
-        <p id="Example1-config">
+        <p id="Example-config">
           For example, the following configuration fragment directs the
           Checker to not report audit events with severity
           level <code>info</code>:
         </p>
         <source>
 &lt;module name=&quot;Checker&quot;&gt;
+  &lt;module name=&quot;TreeWalker&quot;&gt;
```
Comment (romani) [inline src/xdocs/filters/severitymatchfilter.xml:58]: please do us a favor and move this example to "Example of Usage"

File: src/main/resources/com/puppycrawl/tools/checkstyle/filters/messages.properties [hunk 2]
```diff
@@ -0,0 +1 @@
+name.invalidPattern=Name ''{0}'' must match pattern ''{1}''.
```
Comment (romani) [inline src/main/resources/com/puppycrawl/tools/checkstyle/filters/messages.properties:1]: what forcing you to add this file and such property?

**Review by romani — CHANGES_REQUESTED** [review]
items:

**Comment by romani** [comment]
>  So, I added 'parametername' and 'methodname' to the hashmap 'moduleMappings'. I think this approach should be acceptable,

yes, it is defect of our tes suite as it expects modules to be in same package and target module - Filter in this case.

>  So, I added a messages.properties file under src/main/resources/com/puppycrawl/tools/checkstyle/filters, and for subsequent filter example tests, more violation messages can be added to this file to resolve this issue. Is my solution feasible?

not good, can we update internal suite code to take messages from file we will provide as hardcoded values, same as we did for fully qualified names ( item above)

File: src/main/resources/com/puppycrawl/tools/checkstyle/filters/messages.properties (hunk 2 above)
Comment (romani) [reply src/main/resources/com/puppycrawl/tools/checkstyle/filters/messages.properties:1]: see guidance at https://github.com/checkstyle/checkstyle/pull/14784#issuecomment-2049962607

Comment (Kexin2000) [reply src/main/resources/com/puppycrawl/tools/checkstyle/filters/messages.properties]: done

File: src/xdocs/filters/severitymatchfilter.xml (hunk 1 above)
Comment (romani) [reply src/xdocs/filters/severitymatchfilter.xml:58]: I was confused, I recall this

File: src/xdocs/filters/severitymatchfilter.xml [hunk 3]
```diff
@@ -55,12 +55,29 @@
...
         </source>
+        <p id="Example1-code">
+          Code Example:
+        </p>
+        <source>
+public class Example1 {
+
+  public void method1(int V1){} // ok, ParameterNameCheck's severity is info
+
+  public void Method2(){} // violation
```
Comment (romani) [inline src/xdocs/filters/severitymatchfilter.xml:78]: `// violation, as default severity is error`

File: src/xdocs/filters/severitymatchfilter.xml [hunk 4]
```diff
@@ -55,12 +55,29 @@
...
   &lt;/module&gt;
 &lt;/module&gt;
         </source>
+        <p id="Example1-code">
+          Code Example:
+        </p>
+        <source>
+public class Example1 {
+
+  public void method1(int V1){} // ok, ParameterNameCheck's severity is info
```
Comment (romani) [inline src/xdocs/filters/severitymatchfilter.xml:76]: `// ok, ParameterName's severity is info`

File: src/test/java/com/puppycrawl/tools/checkstyle/AbstractModuleTestSupport.java [hunk 5]
```diff
@@ -629,12 +629,21 @@ private static String getMessageBundle(String className) {
...
+                "com.puppycrawl.tools.checkstyle.checks.naming.messages");
+
         if (endIndex < 0) {
             messageBundle = messages;
         }
         else {
             final String packageName = className.substring(0, endIndex);
-            messageBundle = packageName + "." + messages;
+            if ("com.puppycrawl.tools.checkstyle.filters".equals(packageName)) {
+                messageBundle = messageBundleMappings.get(className.substring(endIndex + 1));
```
Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/AbstractModuleTestSupport.java:642]: Please change logic to use value from manual mapping if mapping is present.
If present in special map then use it , else use packageName

**Review by romani — CHANGES_REQUESTED** [review]
Items

**Comment by romani** [comment]
@Kexin2000 ,please find time to finish PR

**Comment by romani** [comment]
@Kexin2000 ,please find time to finish PR.
Let us know if somebody else can finish this PR

**Comment by romani** [comment]
will be done at https://github.com/checkstyle/checkstyle/pull/15021

**Pull request closed without merging**